samcli/commands/build/build_context.py

```diff
@@ -83,6 +84,13 @@ def __exit__(self, *args):
     def _setup_build_dir(build_dir, clean):
         build_path = pathlib.Path(build_dir)
 
+        if os.path.abspath(str(build_path)) == os.path.abspath(str(pathlib.Path.cwd())):
+            exception_message = "'build-dir' should not be the current workding directoy as the directory " \
```

Comment:
can we instead skip the "clean" if build-dir == cwd? It could give to a more intuitive experience

Reply:
Not sure I completely understand. My current understanding of the question is:

"Instead of always doing a "clean" build, we would not do a clean build if build-dir == cwd"?

If that is a fair rewording, then I don't think we should go down this route. Having a conditional do clean or not base upon a parameter that does not relate to a 'clean build' is not intuitive at all. This will make supporting clean builds more difficult in the future.